src/test/java/io/pinecone/clients/ConnectionsMapTest.java

```diff
@@ -0,0 +1,134 @@
+
+    @Test
+    public void testMultiplePineconeInstancesForIndexes() {
+        String host = "some-host";
+        ManageIndexesApi manageIndexesApi = Mockito.mock(ManageIndexesApi.class);
+        IndexModel indexModel = new IndexModel();
+        indexModel.setHost(host);
+
+        PineconeConfig config = new PineconeConfig("testApiKey");
+        config.setHost(host);
+        Pinecone pinecone1 = new Pinecone(config, manageIndexesApi);
+        Pinecone pinecone2 = new Pinecone(config, manageIndexesApi);
+
+        when(pinecone1.describeIndex("index1")).thenReturn(indexModel);
+        when(pinecone1.describeIndex("index2")).thenReturn(indexModel);
+        when(pinecone2.describeIndex("index1")).thenReturn(indexModel);
+        when(pinecone2.describeIndex("index2")).thenReturn(indexModel);
+
+        Index index1Pinecone1 = pinecone1.getIndexConnection("index1");
+        Index index1Pinecone2 = pinecone1.getIndexConnection("index2");
+        Index index2Pinecone1 = pinecone2.getIndexConnection("index1");
+        Index index2Pinecone2 = pinecone2.getIndexConnection("index2");
+        assertEquals(pinecone1.getConnectionsMap().size(), 2);
+        assertEquals(pinecone2.getConnectionsMap().size(), 2);
+
+        index1Pinecone1.close();
+        index2Pinecone1.close();
+        index1Pinecone2.close();
+        index2Pinecone2.close();
+    }
+
+    @Test
+    public void testMultiplePineconeInstancesForAsyncIndexes() {
+        String host = "some-host";
+        ManageIndexesApi manageIndexesApi = Mockito.mock(ManageIndexesApi.class);
+        IndexModel indexModel = new IndexModel();
+        indexModel.setHost(host);
+
+        PineconeConfig config = new PineconeConfig("testApiKey");
+        config.setHost(host);
+        Pinecone pinecone1 = new Pinecone(config, manageIndexesApi);
+        Pinecone pinecone2 = new Pinecone(config, manageIndexesApi);
+
+        when(pinecone1.describeIndex("index1")).thenReturn(indexModel);
+        when(pinecone1.describeIndex("index2")).thenReturn(indexModel);
+        when(pinecone1.describeIndex("index1")).thenReturn(indexModel);
+        when(pinecone1.describeIndex("index2")).thenReturn(indexModel);
+
+        AsyncIndex asyncIndex1Pinecone1 = pinecone1.getAsyncIndexConnection("index1");
+        AsyncIndex asyncIndex2Pinecone1 = pinecone1.getAsyncIndexConnection("index2");
+        AsyncIndex asyncIndex1Pinecone2 = pinecone2.getAsyncIndexConnection("index1");
+        AsyncIndex asyncIndex2Pinecone2 = pinecone2.getAsyncIndexConnection("index2");
+        assertEquals(pinecone2.getConnectionsMap().size(), 2);
+        assertEquals(pinecone2.getConnectionsMap().size(), 2);
+
+        asyncIndex1Pinecone1.close();
+        asyncIndex2Pinecone1.close();
+        asyncIndex1Pinecone2.close();
+        asyncIndex2Pinecone2.close();
+    }
+
+    @Test
+    public void testConnectionRemovedOnIndexClose() throws ApiException {
+        String host = "some-host";
```

Comment:
Nit, but I'd stick to camelcase for the var values. The hyphens seem Python-y in a weird way

Reply:
Ideally we try to mimic the behavior of how the variable's value will look like but since these are values for unit test variables, it won't matter much